ingest-app/src/cmr/ingest/services/jobs_subscriptions.clj

```diff
@@ -0,0 +1,199 @@
+        end-time (last parts)]
+    (assoc raw :start-time start-time :end-time end-time)))
+
+(defn- send-update-subscription-notification-time!
+  "handle any packaging of data here before sending it off to transmit package"
+  [context data]
+  (debug "send-update-subscription-notification-time with" data)
+  (search/save-subscription-notification-time context data))
+
+(defn- filter-gran-refs-by-subscriber-id
+  "Takes a list of granule references and a subscriber id and removes any granule that the user does
+   not have read access to."
+  [context gran-refs subscriber-id]
+  (when (seq gran-refs)
+    (let [permissions (json/parse-string
+                       (access-control/get-permissions context {:user_id subscriber-id
+                                                                :concept_id (map :concept-id gran-refs)}))]
+      (filter (fn [granule-reference]
+                (some #{"read"} (get permissions (:concept-id granule-reference))))
+              gran-refs))))
+
+(defn- subscription->time-constraint
+  "Create a time-constraint from a subscriptions last-notified-at value or amount-in-sec from the end."
+  [subscription end amount-in-sec]
+  {:post [(spec/valid? ::time-constraint %)]}
+  (let [begin (if-let [start (get-in subscription [:extra-fields :last-notified-at])]
+                start
+                (t/minus end (t/seconds amount-in-sec)))]
+    (str begin "," end)))
+
+(defn- search-gran-refs-by-collection-id
+  [context params1 params2 sub-id]
+  (try
+    (let [gran-refs1 (search/find-granule-references context params1)
+          gran-refs2 (search/find-granule-references context params2)]
+      (distinct (concat gran-refs1 gran-refs2)))
+    (catch Exception e
+      (error "Exception caught processing subscription" sub-id " searching with params "
+             (dissoc params1 :token) (dissoc params2 :token) "\n\n" (.getMessage e) "\n\n" e)
+      [])))
+
+(defn- process-subscriptions
+  "Process each subscription in subscriptions into tuple for testing purposes and to use as
+   the input when sending the subscription emails."
+  [context subscriptions]
+  (for [raw-subscription subscriptions
+        :let [time-constraint (subscription->time-constraint
+                               raw-subscription
+                               (t/now)
+                               (email-subscription-processing-lookback))
+              subscription (add-updated-since raw-subscription time-constraint)
+              subscriber-id (get-in subscription [:extra-fields :subscriber-id])
+              sub-id (get subscription :concept-id)
+              coll-id (get-in subscription [:extra-fields :collection-concept-id])
+              query-string (-> (:metadata subscription)
+                               (json/decode true)
+                               :Query)
+              query-params (create-query-params query-string)
+              params1 (merge {:created-at time-constraint
```

Comment:
There is no need to search for `created-at` as `revision-date` alone should be able to cover the use case.

Reply:
wow, this made it through so many reviews.  :) Let me give it a try and see what it does to my tests.